### ISSUE TITLE
Should check nested functions

### ISSUE THREAD

**Issue by PeterJCLaw** [facebook/pyre-check#61]
`python`: 3.5.2, 3.6.5
`pyre`: 0.0.7 (from `pip list`)

Given a file `thing.py`:
``` python
def outer() -> str:
    def inner(x: int, y: str) -> str:
        return x
    return inner(42, None)

outer()

# Copy of 'inner' to prove that it errors at the top level
def outer_inner(x: int, y: str) -> str:
    return x

outer_inner(42, None)
```

I'd expect to get errors from the bad types in the nested `inner` function. For reference, here's what `mypy` (0.600) outputs:

```
$ mypy thing.py 
thing.py:3: error: Incompatible return value type (got "int", expected "str")
thing.py:4: error: Argument 2 to "inner" has incompatible type "None"; expected "str"
thing.py:10: error: Incompatible return value type (got "int", expected "str")
thing.py:12: error: Argument 2 to "outer_inner" has incompatible type "None"; expected "str"
```

Yet while `pyre` does not find errors in the top level equivalent function `outer_inner`, it doesn't emit any errors about the nested function:

```
$ pyre check
 ƛ Found 1 type error!
thing.py:2:4 Incompatible return type [7]: Expected `str` but got `int`.
```

**Comment by dark** [comment]
Hi @PeterJCLaw, thanks for the report.
We are aware of this limitation, and adding support to type-check nested functions is in our todo list.

**Comment by PeterJCLaw** [comment]
@dark Thanks! Is there a list of known limitations somewhere, to save us reporting (and you having to triage) known limitations?

**Comment by dark** [comment]
Unfortunately we don't have such a list in the repo. We should add one.

**Comment by dark** [comment]
This feature has been implemented a while ago, I had forgot to ping this task. The example above now triggers errors on all relevant lines.

**Comment by PeterJCLaw** [comment]
Nice, thanks for fixing 😊
